cmd/thanos/rule.go

```diff
@@ -922,6 +923,12 @@ func queryFuncCreator(
 		}
 
 		return func(ctx context.Context, qs string, t time.Time) (promql.Vector, error) {
+			// See prometheus rules/group.go to see how this is populated
```

Comment:
Probably not a comment we need here?

Reply:
Let's actually add comment to point to where this value is passed instead https://github.com/prometheus/prometheus/blob/main/rules/group.go#L181

Reply:
Pointed to specfic line in comment `// This context value is set by Prometheus. For details on how it's populated, see:https://github.com/prometheus/prometheus/blob/main/rules/group.go#L181`

Reply:
I don't think that erroring here is a good strategy as you cannot really recover from that error as the context is set by Prometheus libraries. If you get an error once your deployment of ruler is pretty much borked. I would rather set empty string if needs be but also we can probably rely on Prometheus to set the expected context.